glib-macros/src/lib.rs

```diff
@@ -310,3 +311,68 @@ pub fn gflags(attr: TokenStream, item: TokenStream) -> TokenStream {
     let gen = gflags_attribute::impl_gflags(&input, &gtype_name);
     gen.into()
 }
+
+/// Macro for deriving implementations of [`glib::clone::Downgrade`] and
+/// [`glib::clone::Upgrade`] traits and a weak type.
```

Comment:
Are those traits not re-exported from the crate root, i.e. `glib::Downgrade` instead of `glib::clone::Downgrade`?

Reply:
if not then maybe we should do that instead, and also include them in the prelude :)

Reply:
So, `glib::clone!` should also be included in the prelude.

Reply:
Macros are always exported from the root of the crate

Reply:
Assuming `derive(Downgrade)` covers most of scenarios (he-he), users barely need to import `Downgrade/Upgrade` traits.

Reply:
But `glib::clone` is not in the prelude. It would be strange if `glib::Downgrade` derive macro will be in the prelude and `glib::clone` not.

Reply:
That's not possible with macros, they're always at the crate root

Reply:
I added macros to the prelude.

But I didn't add traits. They cause a conflict in `glib/src/subclass/object.rs`. Should I also change `subclass/object.rs`?

Reply:
> I added macros to the prelude.

What difference does that make in practice, given that macros are always exported at the crate root?

> But I didn't add traits. They cause a conflict in glib/src/subclass/object.rs. Should I also change subclass/object.rs?

What conflict does it cause?

Reply:
I causes ambiguity between `glib::object::ObjectExt::downgrade` and `glib::clone::Downgrade::downgrade`.

Reply:
I see, let's keep it the way it is then. Please also don't put the macro into the prelude, thanks!

Reply:
I reverted changes in the prelude.